### PR TITLE
New dist build system.

### PR DESCRIPTION
**Pull request by evert**
The following changes completely moves the distribution-creation system to the makefile, using mostly gnu commands.

This produces a 6.1MB zip file for the current version, which is the smallest release yet =)

Unfortunately no .phar, but I think we can live with that.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -113,3 +114,23 @@ test: devinstall-server
 			--dry-run \
 			--diff \
 			.
+
+DISTFILES = .bowerrc CHANGELOG.md LICENSE Makefile README.md bin bootstrap.php bower.json composer.json composer.lock data lib package.json public resource semantic.json tests vendor
+DISTPOINTLESS = .bowerrc Makefile bower.json composer.json composer.lock node_modules tests
+KATANA_VERSION = $(shell php -r "include 'lib/Version.php'; echo Sabre\Katana\Version::VERSION;")
```
Comment (Hywan) [inline Makefile]: I guess we can add a `katana version` command here. Would be clever. Thoughts?

**Comment by Hywan** [comment]
You can also use the `katana stub --zip` and `katana stub --phar` command to produce a ZIP or a PHAR. All you need to do is to create a clean environment.

File: Makefile (hunk 1 above)
Comment (Hywan) [inline Makefile]: Actually we already have `katana --version`.

Comment (evert) [inline Makefile]: Yep that works for me =)

**Comment by evert** [comment]
I don't really see the benefit of invoking a php application for this. `zip` is available everywhere, so why not just use that?

The problem with the stub commands are that they currently don't work unless dev dependencies are installed using composer. And if those dependencies are installed, then the package becomes bigger than it has to be. The phar command gave me a different error altogether, and I couldn't figure out what it meant.

So I think this is an improvement. If we take out those commands altogether it means that we lose several hundred lines of code and several dependencies and replace it with a single `zip` command.

**Comment by Hywan** [comment]
You mean the current ZIP archive does not include dependencies?

**Comment by evert** [comment]
No, sorry.. earlier on I was not able to make a build without `--dev` (development dependencies). Now I can.

**Pull request merged**